### PR TITLE
Adding check box containing group to connect median and a combo box containing the summaries on the Describe > Specific >Box plot

### LINKED ISSUE #5236: Adding a line to connect medians in a boxplot
I am trying to produce the following graph - which is easy in the old Instat - as shown below:
image

This is a single set of boxplots with a line connecting the medians.
In R-Instat I first got the boxplots.
Then added a layer.  I used stat_summary
Then I use the layer parameters.  I make the fun.y the **median** and the geom the **line**
So far so good.  
Then I press return and I have the 2 layers.
Then I return again and try the plot.  It doesn't seem to work.
Now I return to the dialogue and find the second layer doesn't exist any more.
(In fact I later find that after the first Return, I then opt to edit the second layer and it has reset the layer parameters.)
What is wrong or what am I doing wrong?
Note: Even though the stat.summary layer is apparently no longer there, when I run again I still get the stat.summary in the R command.  I need to do a reset to get rid of the code..

### PR DESCRIPTION
**Pull request by Wycklife**
Fixes #5236

@africanmathsinitiative/developers In progress

### CODE REVIEW

**Comment by Wycklife** [comment]
@africanmathsinitiative/developers This is now ready for review.

**Comment by Wycklife** [comment]
@Ogik99 Could you please review this?

**Comment by Ogik99** [comment]
> @Ogik99 Could you please review this?

On it

**Comment by Ogik99** [comment]
Line connecting the median and mean works well. However no line is displayed for both the upper and lower quartiles. See below an example of the lower quartile:
lower_quartile

**Comment by Wycklife** [comment]
@Ogik99 Could please check whether the changes I've resolve the issue?

**Comment by Ogik99** [comment]
@Wycklife  I have reviewed once more but the lines for both the lower and upper quartiles are not displayed when I use the group to connect check box.
The lines however appear when I use another approach i.e. go to plot options, layers, and then add stat summary and finally check fun.y. 
For consistency, is it possible to have the lower and upper quartile options in the group to connect checkbox behave like the mean and median options.

**Comment by dannyparsons** [comment]
@Ogik99 it's useful to always provide the R code when testing and reporting a problem. This will allow you to also say whether the problem is largely with the dialog or the R code. i.e. is `stat_summary` being added to the R code but not displaying (R code problem) or is it not being added at all (dialog problem).
In general try to be as technical as possible when testing. When you notice a problem through testing you should look at the code changes in the pull request to try to see possible causes.
In any case, a review should always be a review of the code as well of how it works.

**Comment by Ogik99** [comment]
> @Ogik99 it's useful to always provide the R code when testing and reporting a problem. This will allow you to also say whether the problem is largely with the dialog or the R code. i.e. is `stat_summary` being added to the R code but not displaying (R code problem) or is it not being added at all (dialog problem).
> In general try to be as technical as possible when testing. When you notice a problem through testing you should look at the code changes in the pull request to try to see possible causes.
> In any case, a review should always be a review of the code as well of how it works.

@dannyparsons this is well noted. Thank you

**Comment by Muthenya** [comment]
@Wycklife what's the state of this?

**Comment by Wycklife** [comment]
@Muthenya The required changes have been made and the pull request is ready to be reviewed

**Comment by maxwellfundi** [comment]
Great, @rdstern  @dannyparsons this is now ready and works very well.

**Comment by maxwellfundi** [comment]
@dannyparsons  These line connectors have been implemented and ready for you.

File: instat/ucrGeom.vb [hunk 1]
```diff
@@ -174,7 +174,7 @@ Public Class ucrGeom
 
         'These are the list of parameters on the documentation, some may have more options than what we have here which am yet to find out
         clsgeom_statsummary.AddLayerParameter("fun.data", "editablelist", Chr(34) & "mean_cl_boot" & Chr(34), lstParameterStrings:={Chr(34) & "mean_sdl" & Chr(34), Chr(34) & "mean_cl_boot" & Chr(34)})
-        clsgeom_statsummary.AddLayerParameter("fun.y", "editablelist", Chr(34) & "mean" & Chr(34), lstParameterStrings:={Chr(34) & "median" & Chr(34), Chr(34) & "mean" & Chr(34)})
+        clsgeom_statsummary.AddLayerParameter("fun.y", "editablelist", Chr(34) & "mean" & Chr(34), lstParameterStrings:={Chr(34) & "median" & Chr(34), Chr(34) & "mean" & Chr(34), Chr(34) & "lower quartile" & Chr(34), Chr(34) & "upper quartile" & Chr(34)})
         clsgeom_statsummary.AddLayerParameter("fun.ymin", "editablelist", Chr(34) & "min" & Chr(34), lstParameterStrings:={Chr(34) & "min" & Chr(34)})
```
Comment (maxwellfundi) [inline instat/ucrGeom.vb]: @Wycklife  was this needed to be added here?

**Comment by maxwellfundi** [comment]
@dannyparsons  I am merging this to add changes required in #4540

**Pull request merged**